source/nanoFramework.Json/JsonConvert.cs

```diff
@@ -175,31 +193,73 @@ private static object PopulateObject(JsonToken rootToken, Type rootType, string
 								memberPath = memberPath + '/' + memberProperty.Name;    // Need to add a slash before appending rootElementType
 							}
 							var memberObject = PopulateObject(memberProperty.Value, memberType, memberPath);
-							memberSetMethod.Invoke(rootInstance, new object[] { memberObject });
-						}
+							if (memberIsProperty) 
+							{
+								memberPropSetMethod.Invoke(rootInstance, new object[] { memberObject });
+							} 
+							else 
+							{
+								memberFieldInfo.SetValue(rootInstance, memberObject);
+							}
+							DebugHelper.DisplayDebug($"{debugIndent}     successfully initialized member {memberProperty.Name} to memberObject");
+						} 
 						else if (memberProperty.Value is JsonValue)
 						{
 							// Don't need any more info - populate the member using memberSetMethod.Invoke()
 							DebugHelper.DisplayDebug($"{debugIndent}     memberProperty.Value is JValue");
-							if (memberType != typeof(DateTime))
+							if (memberType != typeof(DateTime)) 
 							{
-								memberSetMethod.Invoke(rootInstance, new object[] { ((JsonValue)memberProperty.Value).Value });
-							}
-							else
+								DebugHelper.DisplayDebug($"{debugIndent}     attempting to set rootInstance by invoking this member's set method for properties  or  SetValue() for fields");
+								if (((JsonValue)memberProperty.Value).Value == null) 
+								{
+									// This doesn't work for float members that have a value of NaN - check for this and handle it separately
+									DebugHelper.DisplayDebug($"{debugIndent}     memberProperty.Value is null");
+									if (memberIsProperty)
+									{
+										memberPropSetMethod.Invoke(rootInstance, new object[] { null });
+									} 
+									else 
+									{
+										object obj = null;
+										memberFieldInfo.SetValue(rootInstance, obj);
+									}
+									DebugHelper.DisplayDebug($"{debugIndent}     successfully initialized member {memberProperty.Name}  to  null");
+								} 
+								else 
+								{
+									if (memberIsProperty) 
+									{
+										memberPropSetMethod.Invoke(rootInstance, new object[] { ((JsonValue)memberProperty.Value).Value });
+									} 
+									else 
+									{
+										memberFieldInfo.SetValue(rootInstance, ((JsonValue)memberProperty.Value).Value);
+									}
+									DebugHelper.DisplayDebug($"{debugIndent}     successfully initialized member {memberProperty.Name}  to  {((JsonValue)memberProperty.Value).Value} ");
+								}
+							} else 
 							{
 								DateTime dt;
 								var sdt = ((JsonValue)memberProperty.Value).Value.ToString();
-								if (sdt.Contains("Date("))
+								if (sdt.Contains("Date(")) 
 								{
 									dt = DateTimeExtensions.FromASPNetAjax(sdt);
-								}
-								else
+								} 
```

Comment:
Please try and remove any whitespace changes, they make reviewing a lot harder

Reply:
Sorry, this was an attempt to conform to the coding standard.